nixos/modules/services/misc/languagetool.nix

```
@@ -0,0 +1,82 @@
+{ config, lib, options, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.languagetool;
+in {
+  options.services.languagetool = {
+    enable = mkOption {
+      type = types.bool;
+      example = true;
+      default = false;
+      description = mdDoc ''
+        Start the HTTP LanguageTool server.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8081;
+      example = 8081;
+      description = mdDoc ''
+        Port on which LanguageTool listens.
+      '';
+    };
+
+    public = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = mdDoc ''
+        If `true` allow the server to be connected from anywhere,
+        otherwise only from the computer it was started from.
+      '';
+    };
+
+    allowOrigin = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://my-website.org";
+      description = mdDoc ''
+        Set the Access-Control-Allow-Origin header in the HTTP response,
+        used for direct (non-proxy) JavaScript-based access from browsers.
+        `null` to allow access from all sites.
+      '';
+    };
+
+    configuration = mkOption {
+      type = types.str;
```

```suggestion
      type = types.lines;
```
to allow merging. better yet: use one of the `pkgs.formats` for an RFC 42 settings option if applicable (configuration looks a bit like ini?)

---

It's a Java property file. I'll change to RFC 42 settings.